test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java

```diff
@@ -584,21 +585,18 @@ public void status() throws Exception {
     public void pruneNarrative() throws Exception {
         wsCli1.createWorkspace(new CreateWorkspaceParams()
                 .withWorkspace("narprune"));
-        
+
         final Map<String, Object> data = new HashMap<>();
         data.put("source", "a long string");
```

Comment:
Why is this test changing? Seems like the changes defeat the purpose of the test.

Reply:
Because it was violating the strict mapping rule.

Reply:
Then shouldn't the test should be updated so it doesn't violate that rule but still tests what it's intended to test?

Reply:
That's exactly the change that I made.

Reply:
I'm confused too. I was guessing the point of this test was to make sure that these other key values are dropped from the indexed object.  Why is it violating the strict mapping rule if they are getting dropped?

Reply:
But this test is meant to test the pruning that happens in the search api.

Here
https://github.com/kbase/KBaseSearchEngine/blob/master/lib/src/kbasesearchengine/main/TemporaryNarrativePruner.java

So the idea, as I understand it, is the index record has these keys and we want to make sure the search api is dropping them.  So excluding them from the start doesn't test the pruner.

I'm still confused as to why the strict mapping is breaking things because those should be valid keys.

Reply:
I totally missed the documentation on top of the test about this test belonging to the narrative pruner. I have put back the keys that get pruned to show that the pruning functionality is working. To answer your question about why the strict mapping is breaking this test, it is because the keys in the parsed object need to be backed up with indexing rules as the indexing rules are used to create the mapping. No rule for a key, implies no mapping for that key and this causes the indexing to fail. So now, all I have done is add indexing rules for each of the keys.